### DISCUSSION TITLE
concentrating all of yourself's dependency packages into the ONE package emmett

### DISCUSSION THREAD

**Discussion by ockan** [emmett-framework/emmett#528] [Ideas]
emmett is a great framework,but its most dependency packages are writed by yourself.

maybe concentrating them into the package emmett is a better idea.

**Comment by gi0baro** [comment]
> maybe concentrating them into the package emmett is a better idea.

I strongly disagree.

As a matter of fact, Emmett was indeed a single package, but it was split into the current structure in recent years.
The benefit of this approach makes specific feature-sets available to everyone else, for instance the existence of the Renoir package made it possible to be used by projects like [py4web](https://github.com/web2py/py4web) and [the kivy website](https://github.com/kivy/kivy-website), and projects like [noir](https://github.com/gi0baro/noir) to exists. I can make many more examples, but the principle is the same.

Also, I don't see any benefit/difference to final users, given that `pip install emmett` also take care of all the dependencies involved (which are no more than other frameworks btw).
